### PR TITLE
Add AutoUpdate gRPC service implementation

### PR DESCRIPTION
**Pull request by vapopov**
In this PR implemented gRPC service for auto updates resources (config/version)

### CODE REVIEW

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -5435,6 +5436,16 @@ func NewGRPCServer(cfg GRPCServerConfig) (*GRPCServer, error) {
 	}
 	userprovisioningpb.RegisterStaticHostUsersServiceServer(server, staticHostUserServer)
 
+	autoUpdateServiceServer, err := autoupdatev1.NewService(autoupdatev1.ServiceConfig{
+		Authorizer: cfg.Authorizer,
+		Backend:    cfg.AuthServer.Services,
+		Cache:      cfg.AuthServer.Services,
```
Comment (vapopov) [inline lib/auth/grpcserver.go:5442]: this going to be replaced to `cfg.AuthServer.Cache` in next pr with cache implementation

File: lib/authz/permissions.go [hunk 2]
```diff
@@ -913,6 +913,8 @@ func roleSpecForProxy(clusterName string) types.RoleSpecV6 {
 				types.NewRule(types.KindSAMLIdPServiceProvider, services.RO()),
 				types.NewRule(types.KindUserGroup, services.RO()),
 				types.NewRule(types.KindClusterMaintenanceConfig, services.RO()),
+				types.NewRule(types.KindAutoUpdateConfig, services.RO()),
+				types.NewRule(types.KindAutoUpdateVersion, services.RO()),
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/authz/permissions.go:917]: I'll move this one to client and cache implementation, we need to have this to be able access resources without auth for `find` endpoint

File: lib/auth/autoupdate/autoupdatev1/service.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	return &Service{
+		authorizer: cfg.Authorizer,
+		backend:    cfg.Backend,
+		cache:      cfg.Cache,
+	}, nil
+}
+
+// GetAutoUpdateConfig gets the current autoupdate config singleton.
+func (s *Service) GetAutoUpdateConfig(ctx context.Context, req *autoupdate.GetAutoUpdateConfigRequest) (*autoupdate.AutoUpdateConfig, error) {
+	config, err := s.cache.GetAutoUpdateConfig(ctx)
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/auth/autoupdate/autoupdatev1/service.go:80]: yes

File: lib/auth/autoupdate/autoupdatev1/service.go [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+	if err := s.backend.DeleteAutoUpdateConfig(ctx); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &emptypb.Empty{}, nil
+}
+
+// GetAutoUpdateVersion gets the current autoupdate version singleton.
+func (s *Service) GetAutoUpdateVersion(ctx context.Context, req *autoupdate.GetAutoUpdateVersionRequest) (*autoupdate.AutoUpdateVersion, error) {
+	version, err := s.cache.GetAutoUpdateVersion(ctx)
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/auth/autoupdate/autoupdatev1/service.go:168]: yes

File: lib/auth/autoupdate/autoupdatev1/service.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/auth/autoupdate/autoupdatev1/service.go:80]: We fetch this data for `webapi/find` endpoint, which should be accessible without auth (config and version)

[the remark this answers is not in the document]
Comment (vapopov) [reply lib/auth/autoupdate/autoupdatev1/service.go:80]: we should be able to define any proxy and get information about if autoupdate is enabled and get required version for the tools

https://github.com/gravitational/teleport/pull/39805/files#diff-8d3c0408dd826f27e9f13cd15f09f4d7e89202862f0b13ce131a2e3072c2a40dR199-R213

[the remark this answers is not in the document]
Comment (sclevine) [reply lib/auth/autoupdate/autoupdatev1/service.go:80]: `tctl` should be able to poll the `/webapi/find` endpoint to enable that command. I don't think it needs direct unauthenticated access to this RPC. The proxy should be able authenticate with auth using its own credentials.

[the remark this answers is not in the document]
Comment (vapopov) [reply lib/auth/autoupdate/autoupdatev1/service.go:93]: got it, I've added auth for the getters

**Review by rosstimothy — APPROVED** [review]

**Review by sclevine — APPROVED** [review]

**Pull request merged**